apps/web/app/lib/survey-block-builder.ts

```diff
@@ -0,0 +1,308 @@
+  subheader?: string;
+  required?: boolean;
+  isColorCodingEnabled?: boolean;
+}): TSurveyRatingElement => {
+  return {
+    id: id ?? createId(),
+    type: TSurveyElementTypeEnum.Rating,
+    subheader: subheader ? createI18nString(subheader, []) : undefined,
+    headline: createI18nString(headline, []),
+    scale,
+    range,
+    required: required ?? false,
+    isColorCodingEnabled,
+    lowerLabel: lowerLabel ? createI18nString(lowerLabel, []) : undefined,
+    upperLabel: upperLabel ? createI18nString(upperLabel, []) : undefined,
+  };
+};
+
+export const buildConsentElement = ({
+  id,
+  headline,
+  subheader,
+  label,
+  required,
+}: {
+  id?: string;
+  headline: string;
+  subheader: string;
+  required?: boolean;
+  label: string;
+}): TSurveyConsentElement => {
+  return {
+    id: id ?? createId(),
+    type: TSurveyElementTypeEnum.Consent,
+    subheader: createI18nString(subheader, []),
+    headline: createI18nString(headline, []),
+    required: required ?? false,
+    label: createI18nString(label, []),
+  };
+};
+
+export const buildCTAElement = ({
+  id,
+  headline,
+  subheader,
+  buttonExternal,
+  required,
+  dismissButtonLabel,
+  buttonUrl,
+}: {
+  id?: string;
+  headline: string;
+  buttonExternal: boolean;
+  subheader: string;
+  required?: boolean;
+  dismissButtonLabel?: string;
+  buttonUrl?: string;
+}): TSurveyCTAElement => {
+  return {
+    id: id ?? createId(),
+    type: TSurveyElementTypeEnum.CTA,
+    subheader: createI18nString(subheader, []),
+    headline: createI18nString(headline, []),
+    dismissButtonLabel: dismissButtonLabel ? createI18nString(dismissButtonLabel, []) : undefined,
+    required: required ?? false,
+    buttonExternal,
+    buttonUrl,
+  };
+};
+
+export const buildNPSElement = ({
+  id,
+  headline,
+  subheader,
+  lowerLabel,
+  upperLabel,
+  required,
+  isColorCodingEnabled = false,
+}: {
+  id?: string;
+  headline: string;
+  subheader?: string;
+  lowerLabel?: string;
+  upperLabel?: string;
+  required?: boolean;
+  isColorCodingEnabled?: boolean;
+}): TSurveyNPSElement => {
+  return {
+    id: id ?? createId(),
+    type: TSurveyElementTypeEnum.NPS,
+    subheader: subheader ? createI18nString(subheader, []) : undefined,
+    headline: createI18nString(headline, []),
+    required: required ?? false,
+    isColorCodingEnabled,
+    lowerLabel: lowerLabel ? createI18nString(lowerLabel, []) : undefined,
+    upperLabel: upperLabel ? createI18nString(upperLabel, []) : undefined,
+  };
+};
+
+// Helper function to create block-level jump logic based on operator
+export const createBlockJumpLogic = (
+  sourceElementId: string,
+  targetBlockId: string,
+  operator: "isSkipped" | "isSubmitted" | "isClicked"
+): TSurveyBlockLogic => ({
+  id: createId(),
+  conditions: {
+    id: createId(),
+    connector: "and",
+    conditions: [
+      {
+        id: createId(),
+        leftOperand: {
+          value: sourceElementId,
+          type: "question",
```

Comment:
I think I missed this while changing the types initially, this should also be changed to `element` just like `jumpToQuestion` has also been changed to now `jumpToBlock`, right? There's probably too much code right now with this though and maybe the best way to do this would be when we write the migration, what do you think?